backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/LayoutService.java

```diff
@@ -82,7 +85,47 @@ public LayoutService(ELKDiagramConverter elkDiagramConverter, IncrementalLayoutD
     @Override
     public Diagram layout(IEditingContext editingContext, Diagram diagram) {
         ELKConvertedDiagram convertedDiagram = this.elkDiagramConverter.convert(diagram);
+        return layoutInternal(editingContext, diagram, convertedDiagram);
+    }
+
+    @Override
+    public Node layoutNode(IEditingContext editingContext, Diagram diagram, Node node) {
+        ELKConvertedDiagram convertedDiagram = this.elkDiagramConverter.convert(diagram, node);
+        Diagram layoutedDiagram = layoutInternal(editingContext, diagram, convertedDiagram);
+
+        return layoutedDiagram.getNodes()
+                .stream()
+                .filter(layoutedDiagramNode -> layoutedDiagramNode.getId().equals(node.getId()))
+                .findFirst().orElse(null);
+    }
 
+    @Override
+    public Diagram incrementalLayout(IEditingContext editingContext, Diagram newDiagram, Optional<IDiagramEvent> optionalDiagramElementEvent) {
+        List<Node> nodes = newDiagram.getNodes();
+        for (int i = 0, nodesSize = nodes.size(); i < nodesSize; i++) {
+            Node node = nodes.get(i);
+            if (node.getPosition() == Position.UNDEFINED) {
+                Node element = layoutNode(editingContext, newDiagram, node);
+                nodes.set(i, Node.newNode(element).position(Position.UNDEFINED).build());
+            }
+        }
```

Comment:
Relying on the ELK layout to improve the incremental layout is not a good approach. It would be a major shift in our layout philosophy and would thus require a lot of discussion beforehand. It is also way overkill for that specific issue and it would be a major performance hit. The proper fix should be to improve the incremental layout algorithm in question.

I don't think that the approach taken by this PR can be salvaged.